crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -384,26 +387,59 @@ impl Segment3d {
     /// Panics if `point1 == point2`
     #[inline(always)]
     pub fn from_points(point1: Vec3, point2: Vec3) -> (Self, Vec3) {
-        let diff = point2 - point1;
-        let length = diff.length();
+        let segment = Self::new(point1, point2);
 
-        (
-            // We are dividing by the length here, so the vector is normalized.
-            Self::new(Dir3::new_unchecked(diff / length), length),
-            (point1 + point2) / 2.,
-        )
+        (segment, segment.center())
```

Comment:
This should be deprecated like in 2D. Also, why is the implementation here slightly different?